src/Symfony/Component/Serializer/Encoder/CsvEncoder.php

```diff
@@ -101,15 +92,15 @@ public function encode(mixed $data, string $format, array $context = []): string
         $endOfLine = $context[self::END_OF_LINE] ?? $this->defaultContext[self::END_OF_LINE];
 
         if (!($context[self::NO_HEADERS_KEY] ?? $this->defaultContext[self::NO_HEADERS_KEY])) {
-            fputcsv($handle, $headers, $delimiter, $enclosure, $escapeChar);
+            fputcsv($handle, $headers, $delimiter, $enclosure);
```

Comment:
you need to pass an empty string to `fputcsv`, as this is not the default behavior of `fputcsv` (the default behavior being a deprecated one)